api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/payments/PaymentService.java

```diff
@@ -37,19 +38,30 @@
     private PaymentCalculator paymentCalculator;
     private ReportPaymentMessageSender reportPaymentMessageSender;
 
+    /**
+     * Calculates the amount to pay, based on the total amount from the voucher entitlement, taking into account the balance currently on the card.
+     *
+     * @param paymentCycle  the current payment cycle, which should have a voucher entitlement.
+     * @return the amount that should be deposited on the card account.
+     */
+    public PaymentCalculation calculatePaymentAmount(PaymentCycle paymentCycle) {
+        CardBalanceResponse balance = cardClient.getBalance(paymentCycle.getClaim().getCardAccountId());
+        return paymentCalculator.calculatePaymentCycleAmountInPence(paymentCycle.getVoucherEntitlement(), balance.getAvailableBalanceInPence());
+    }
+
     /**
      * Method used to store a failed payment to the database with the status of FAILURE. Will attempt to
      * retrieve the payment amount and reference from the failure event if available.
      *
      * @param paymentCycle  The payment cycle for the payment
-     * @param cardAccountId The card account id
      * @param failureEvent  The failure event.
      */
-    public void saveFailedPayment(PaymentCycle paymentCycle, String cardAccountId, FailureEvent failureEvent) {
+    public void saveFailedPayment(PaymentCycle paymentCycle, FailureEvent failureEvent) {
```

Comment:
As a general observation on this class, the majority of its lines of code are building POJOs, can we refactor any/all of them into builders/factories or even static constructor methods on those classes? This class would shrink down dramatically then.

Reply:
Added a TODO to create a factory.